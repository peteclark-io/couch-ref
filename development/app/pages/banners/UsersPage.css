@import "../../css/branding.css";

.splash {
   display: flex;
   flex-flow: column;
   width: 100%;
   height: 100%;
   background-color: var(--brand-primary);
   text-align: center;
   color: var(--brand-secondary);
}

.brand {
   font-family: var(--font-primary);
   font-size: 6em;
   margin-top: 6vh;
   flex: 0 1 auto;
}

.back {
   text-align: left;
   color: white;
   margin-bottom: 16px;
}
