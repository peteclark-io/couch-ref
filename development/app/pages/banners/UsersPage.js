'use strict';

import React from 'react';
import {connect} from 'react-redux';

import styles from './UsersPage.css';

const UsersPage = React.createClass({

  propTypes: {
    user: React.PropTypes.object,
    clubs: React.PropTypes.array
  },

  render: function() {
    if (!this.props.user.remote || !this.props.clubs){
      return null;
    }

    return (
      <div className={styles.splash}>
         <h1 className={styles.brand}>Tell Us About Yourself!</h1>
         <h3 className={styles.question}>Who Do You Support?</h3>

         <div className={styles['clubs-container']}>
           <ul className={styles.clubs}>
              {this.props.clubs.map((c) => {
                return (
                  <li key={c.name}>
                    <img src={c.crestUrl} alt={c.name} onClick={() => {
                      console.log('click!')
                    }}></img>
                  </li>
                );
              })}
           </ul>
         </div>

      </div>
    );
  }
});


const getUser = (state = {user: {}}, id) => {
  return state.user.remote ? state.user : null;
};

const getClubs = (state = {clubs: []}, id) => {
  return state.clubs.length > 0 ? state.clubs : null;
};

const mapStateToProps = (state) => {
   return {
     user: getUser(state),
     clubs: getClubs(state)
   };
};

const mapDispatchToProps = (dispatch) => {
  return {
    vote: (question, result) => {
      saveVote(question, result);
      dispatch(vote(question, result));
    }
  }
};

const LiveUsersPage = connect(
  mapStateToProps
)(UsersPage);

export default LiveUsersPage;
