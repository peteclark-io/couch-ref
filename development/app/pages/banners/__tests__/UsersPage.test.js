'use strict';

import React from 'react';
import UsersPage from '../UsersPage.js';
import renderer from 'react-test-renderer';

test('UsersPage shouldn\'t change much.', () => {
   const component = renderer.create(
      <UsersPage children={<h1>hi!</h1>} />
   );

   let tree = component.toJSON();
   expect(tree).toMatchSnapshot();
});
