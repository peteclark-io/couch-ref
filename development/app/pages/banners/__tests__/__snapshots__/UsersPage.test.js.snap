exports[`test UsersPage shouldn't change much. 1`] = `
<div
  className="splash">
  <h1>
    hi!
  </h1>
</div>
`;
